include/envoy/stream_info/filter_state.h

```diff
@@ -87,6 +87,11 @@ class FilterState {
             (dynamic_cast<const T*>(getDataReadOnlyGeneric(data_name)) != nullptr));
   }
 
+  /**
+   * @return a map of data names and their corresponding objects.
+   */
+  const std::map<std::string, std::unique_ptr<FilterObject> getRawState() const PURE;
+
```

Comment:
So in the absence of a well known namespace, given that we have things like MetadataMatcher etc that rely on iterating through the map, I hope this interface is okay. If this is too much of a blasphemy for the filter state's strict write-once semantics, then I can split the write-many store out into a separate map inside filter state and return that map directly.